### DISCUSSION TITLE
Help: GFS DSWRF

### DISCUSSION THREAD

**Discussion by peterdudfield** [blaylockbk/Herbie#240] [Q&A] [answered]
Hi

Thanks so much for this repo, I think its really good. I'm trying to get some solar irradence data from GFS, but I cant seem to find any. 

```
from herbie import Herbie

H = Herbie("2023-10-12 12:00", model="gfs")
H.inventory().to_csv('gfs.csv')
```
But then I cant find anything to do with solar irradanece, I thought it would come up as DSWRF and DLWRF.
Can anyone help?

Sorry if this is not the right place to put this

**Comment by blaylockbk** [comment]
Good question. You're on the right track. I'll have to sit at my computer and dig around, but have you tried looking in of the the different GFS products? Sometimes the different resolutions have different variables.

**Comment by blaylockbk** [comment, marked as answer]
It looks like the `DSWRF` variable doesn't exist for `fxx=0`, but it does for the forecast hours.

```python
from herbie import Herbie
H = Herbie("2023-10-01", model="gfs", fxx=6)
H.inventory("DS")
```
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>grib_message</th>
      <th>start_byte</th>
      <th>end_byte</th>
      <th>range</th>
      <th>reference_time</th>
      <th>valid_time</th>
      <th>variable</th>
      <th>level</th>
      <th>forecast_time</th>
      <th>search_this</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>652</th>
      <td>653</td>
      <td>461198994</td>
      <td>462155002</td>
      <td>461198994-462155002</td>
      <td>2023-10-01</td>
      <td>2023-10-01 06:00:00</td>
      <td>DSWRF</td>
      <td>surface</td>
      <td>0-6 hour ave fcst</td>
      <td>:DSWRF:surface:0-6 hour ave fcst</td>
    </tr>
  </tbody>
</table>

```python
ds = H.xarray("DSWRF")
ds.dswrf.plot()
```

![image](https://github.com/blaylockbk/Herbie/assets/6249613/f2b1cabc-31e7-418e-9a3c-2b40af877962)

**Reply by blaylockbk** [reply]
Same for `DLWRF`

```python
ds = H.xarray("DLWRF")
ds.dlwrf.plot()
```

![image](https://github.com/blaylockbk/Herbie/assets/6249613/046d358c-08d5-4508-9cfa-68ea9877d31d)

**Reply by blaylockbk** [reply]
Thanks for your question, @peterdudfield, and I'm glad you enjoy Herbie 😀

**Reply by blaylockbk** [reply]
Hint: You can use regex to get both longwave and shortwave flux

```python
ds = H.xarray("D[L,S]WRF")
```
<img width="539" alt="image" src="https://github.com/blaylockbk/Herbie/assets/6249613/c943e136-3565-42ed-b6fe-321e021c77a9">

**Comment by peterdudfield** [comment]
Thank you very much @blaylockbk , im happy if you want to close this

**Comment by williamhobbs** [comment]
I have been referring to this NOAA inventory page for double-checking things like this: https://www.nco.ncep.noaa.gov/pmb/products/. 

For example, the GFS page (https://www.nco.ncep.noaa.gov/pmb/products/gfs/) let's you browse a typical f000 file (https://www.nco.ncep.noaa.gov/pmb/products/gfs/gfs.t00z.pgrb2.0p50.f000.shtml) and one for f003-384 (https://www.nco.ncep.noaa.gov/pmb/products/gfs/gfs.t00z.pgrb2.0p50.f003.shtml). A Ctrl-F search for "DSWRF" on the first one finds nothing, but it shows up in the second one: 

![image](https://github.com/blaylockbk/Herbie/assets/45701090/12971fad-4279-4069-bf4e-a3b4b6d23bed)
